peerconnection_media_test.go

```diff
@@ -1036,63 +1036,75 @@ func TestGetRegisteredRTPCodecs(t *testing.T) {
+	runTest := func(trackCount int) {
+		addSingleTrack := func(p *PeerConnection) *Track {
+			track, err := p.NewTrack(DefaultPayloadTypeVP8, rand.Uint32(), fmt.Sprintf("video-%d", rand.Uint32()), fmt.Sprintf("video-%d", rand.Uint32()))
+			assert.NoError(t, err)
 
-		_, err = p.AddTrack(track)
-		assert.NoError(t, err)
+			_, err = p.AddTrack(track)
+			assert.NoError(t, err)
 
-		return track
-	}
+			return track
+		}
 
-	lim := test.TimeOut(time.Second * 30)
-	defer lim.Stop()
+		pcOffer, err := NewPeerConnection(Configuration{SDPSemantics: SDPSemanticsPlanB})
+		assert.NoError(t, err)
 
-	report := test.CheckRoutines(t)
-	defer report()
+		pcAnswer, err := NewPeerConnection(Configuration{SDPSemantics: SDPSemanticsPlanB})
+		assert.NoError(t, err)
 
-	pcOffer, err := NewPeerConnection(Configuration{SDPSemantics: SDPSemanticsPlanB})
-	assert.NoError(t, err)
+		var onTrackWaitGroup sync.WaitGroup
+		onTrackWaitGroup.Add(trackCount)
+		pcAnswer.OnTrack(func(track *Track, r *RTPReceiver) {
+			onTrackWaitGroup.Done()
+		})
 
-	pcAnswer, err := NewPeerConnection(Configuration{SDPSemantics: SDPSemanticsPlanB})
-	assert.NoError(t, err)
+		done := make(chan struct{})
+		go func() {
+			onTrackWaitGroup.Wait()
+			close(done)
+		}()
 
-	var onTrackWaitGroup sync.WaitGroup
-	onTrackWaitGroup.Add(2)
-	pcAnswer.OnTrack(func(track *Track, r *RTPReceiver) {
-		onTrackWaitGroup.Done()
-	})
+		_, err = pcAnswer.AddTransceiverFromKind(RTPCodecTypeVideo)
+		assert.NoError(t, err)
 
-	done := make(chan struct{})
-	go func() {
-		onTrackWaitGroup.Wait()
-		close(done)
-	}()
+		outboundTracks := []*Track{}
+		for i := 0; i < trackCount; i++ {
+			outboundTracks = append(outboundTracks, addSingleTrack(pcOffer))
+		}
 
-	_, err = pcAnswer.AddTransceiverFromKind(RTPCodecTypeVideo)
-	assert.NoError(t, err)
+		assert.NoError(t, signalPair(pcOffer, pcAnswer))
 
-	track1 := addSingleTrack(pcOffer)
-	track2 := addSingleTrack(pcOffer)
+		func() {
+			for {
+				select {
+				case <-time.After(20 * time.Millisecond):
+					for _, track := range outboundTracks {
+						assert.NoError(t, track.WriteSample(media.Sample{Data: []byte{0x00}, Samples: 1}))
+					}
+				case <-done:
+					return
+				}
+			}
+		}()
 
-	assert.NoError(t, signalPair(pcOffer, pcAnswer))
+		assert.NoError(t, pcOffer.Close())
+		assert.NoError(t, pcAnswer.Close())
+	}
 
-	func() {
-		for {
-			select {
-			case <-time.After(20 * time.Millisecond):
-				assert.NoError(t, track1.WriteSample(media.Sample{Data: []byte{0x00}, Samples: 1}))
-				assert.NoError(t, track2.WriteSample(media.Sample{Data: []byte{0x00}, Samples: 1}))
-			case <-done:
-				return
-			}
-		}
-	}()
+	lim := test.TimeOut(time.Second * 30)
+	defer lim.Stop()
 
-	assert.NoError(t, pcOffer.Close())
-	assert.NoError(t, pcAnswer.Close())
+	report := test.CheckRoutines(t)
+	defer report()
+
+	t.Run("Single Track", func(t *testing.T) {
+		runTest(1)
+	})
+	t.Run("Multi Track", func(t *testing.T) {
+		runTest(2)
```

Comment:
I'd pass `t` as an argument to `runTest` to ensure we get the details about which test case failed in the test logs.

Reply:
Fixed!

Reply:
Fixed!